### PR TITLE
Fix J1939Reader

### PR DESCRIPTION
**Pull request by sophokles73**
The reader erroneously referred to a non-existing (private) member of
the DBCReader. This has been fixed.

Also added some tests for verifying the J1939Reader's functionality.

### CODE REVIEW

**Review by lukasmittag — APPROVED** [review]
Codewise it looks good to me, but couldn't test it @erikbosch can you do it?

File: dbc2val/test/test_dbc/test1_2.kcd [hunk 1]
```diff
@@ -1,5 +1,24 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<NetworkDefinition xmlns="http://kayak.2codeornot2code.org/1.0">
+<!--
+Copyright (c) 2023 Robert Bosch GmbH
```
Comment (erikbosch) [inline dbc2val/test/test_dbc/test1_2.kcd:3]: @SebastianSchildt  - for new files, do we want this form or the more generic "Copyright (c) 2023 Contributors to the Eclipse Foundation"

Comment (sophokles73) [reply dbc2val/test/test_dbc/test1_2.kcd:3]: good point :+1: 

the other Eclipse projects I am involved in usually take the more generic approach and let contributors add their affiliation to the NOTES.md file ...

**Comment by erikbosch** [comment]
> Codewise it looks good to me, but couldn't test it @erikbosch can you do it?

That is part of our current problem for J1939 - we do not have any "real" (or a faked) public log file or signal definitions, so we need to rely on unit tests and on that the contributor has performed more exhaustive end-to-end testing. So we have problems doing end-to-end tests.

**Comment by sophokles73** [comment]
Well, I tried to start alleviating the situation by adding some unit tests for the j1939 processing. For verifying the correctness of the code, IMHO it should make no difference if we use *real* SAE J1939 message/signal definitions or synthetic ones.

I agree, however, that the test suite could be a little more thorough ;-)

File: dbc2val/test/test_dbc/test1_2.kcd (hunk 1 above)
Comment (SebastianSchildt) [reply dbc2val/test/test_dbc/test1_2.kcd:3]: We have for some time now been using the generic "Copyright (c) 2023 Contributors to the Eclipse Foundation" for all new files, as that seems to be the current best practice.
It is however currently neither enforced nor written down as a decision somewhere, we just went where the river took us.

Comment (sophokles73) [reply dbc2val/test/test_dbc/test1_2.kcd:3]: Ok, I will adapt the PR accordingly.

**Comment by sophokles73** [comment]
@erikbosch anything else I need to change?

**Comment by erikbosch** [comment]
> @erikbosch anything else I need to change?

No, looks good to me. I like that you add unit tests.

Concerning testing, it is true that it does not matter if it is real data or faked data, but it would be good to have a j1939 log file + message definition so that we could do a real end-to-end test as part of release testing sanity checks, similar to how we do for the regular included log file for dbcfeeder. Like start dbcfeeder in j1939 mode with a j1939 log file as input and verify that expected values actually are sent to Databroker. But that is more of a long term goal :-)

**Pull request merged**